### ISSUE TITLE
REGR: df.apply('any', axis=1) fails with ValueError

### ISSUE THREAD

**Issue by vnlitvinov** [pandas-dev/pandas#48656]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame({'a': [0, 1]})
print(df.apply('any', axis=1))
```

### Issue Description

When running the snippet above, pandas 1.5 produces the following stack trace:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Vass\.conda\envs\pandas\lib\site-packages\pandas\core\frame.py", line 9558, in apply
    return op.apply().__finalize__(self, method="apply")
  File "C:\Users\Vass\.conda\envs\pandas\lib\site-packages\pandas\core\apply.py", line 720, in apply
    return self.apply_str()
  File "C:\Users\Vass\.conda\envs\pandas\lib\site-packages\pandas\core\apply.py", line 923, in apply_str
    return super().apply_str()
  File "C:\Users\Vass\.conda\envs\pandas\lib\site-packages\pandas\core\apply.py", line 556, in apply_str
    raise ValueError(f"Operation {f} does not support axis=1")
ValueError: Operation any does not support axis=1
```

I believe the root cause is this chunk of code:
https://github.com/pandas-dev/pandas/blob/c68a96f6994a1c4a3e684669b293b5ae37634f2c/pandas/core/apply.py#L550-L556

which should get the signature of `pandas.DataFrame.any` but, as `.any()` is decorated:
https://github.com/pandas-dev/pandas/blob/c68a96f6994a1c4a3e684669b293b5ae37634f2c/pandas/core/generic.py#L11629-L11647

it gets the signature of the _decorator function_ instead.

### Expected Behavior

The snippet above is printing the following when run on pandas 1.4.4:

```
0    False
1     True
dtype: bool
```

I believe it should print the same on 1.5.x, too.

### Installed Versions

<details>
<summary>pd.show_versions()</summary>

```
INSTALLED VERSIONS
------------------
commit           : 87cfe4e38bafe7300a6003a1d18bd80f3f77c763
python           : 3.8.13.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.22000
machine          : AMD64
processor        : AMD64 Family 25 Model 80 Stepping 0, AuthenticAMD
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Russian_Russia.1251

pandas           : 1.5.0
numpy            : 1.23.3
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 65.3.0
pip              : 22.2.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None
```

</details>

**Comment by MarcoGorelli** [comment]
Thanks @vnlitvinov for the report, looks like this was caused by https://github.com/pandas-dev/pandas/pull/44896

**Comment by phofl** [comment]
Yep, can confirm:

```
commit a59582da8c9fa410ac05e25ee117a7fef996247f
Author: Sachin Yadav <40866128+yadav-sachin@users.noreply.github.com>
Date:   Mon Apr 11 01:04:27 2022 +0530

    DEPR: non-keyword arguments in any (#44896)
```

**Comment by MarcoGorelli** [comment]
Hey @vnlitvinov 

Thanks for having reported this - there's now a release candidate for pandas 2.0.0, which you can install with
```
mamba install -c conda-forge/label/pandas_rc pandas==2.0.0rc0
```
or
```
python -m pip install --upgrade --pre pandas==2.0.0rc0
```
If you try it out and report bugs, then we can fix them before the final 2.0.0 release
